.github/workflows/industrial_ci_action.yml

```diff
@@ -10,19 +10,50 @@ jobs:
     strategy:
       matrix:
         env:
-          - {ROS_DISTRO: foxy, ROS_REPO: main}
-          - {ROS_DISTRO: foxy, ROS_REPO: testing}
+          - {
+            CACHE_PREFIX: 'foxy-main-debug-ccov',
+            ROS_DISTRO: foxy, ROS_REPO: main, CCOV_UPLOAD: true,
+            CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS='--coverage' -DCMAKE_CXX_FLAGS='--coverage'",
+            AFTER_RUN_TARGET_TEST: './.ci.prepare_codecov',
+            ADDITIONAL_DEBS: 'lcov'
+          }
+          - {
+            CACHE_PREFIX: 'foxy-testing', ROS_DISTRO: foxy, ROS_REPO: testing, 
+            CCOV_UPLOAD: false
+          }
     env:
       UPSTREAM_WORKSPACE: warehouse_ros_mongo.repos
-      CCACHE_DIR: ~/.ccache
+      AFTER_SETUP_UPSTREAM_WORKSPACE: 'vcs pull $BASEDIR/upstream_ws/src'
+      CCACHE_DIR: "${{ github.workspace }}/.ccache"
+      BASEDIR: ${{ github.workspace }}/.work
```

Comment:
```suggestion
      AFTER_SETUP_UPSTREAM_WORKSPACE: 'vcs pull $BASEDIR/upstream_ws/src'
      CACHE_PREFIX: "${{ matrix.ROS_DISTRO }}-${{ matrix.ROS_REPO }}${{ matrix.CCOV_UPLOAD && '-ccov' || '' }}"
      CCACHE_DIR: "${{ github.workspace }}/.ccache"
      BASEDIR: ${{ github.workspace }}/.work
```

Reply:
I like this, thank you.  Will test using this next.